project/src/main/java/com/google/sps/servlets/BuyServlet.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.sps.data.*;  
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.sql.ResultSet;
+import java.sql.Connection;
+import java.sql.SQLException;
+import javax.sql.DataSource;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * servlet for buying a new investment
+ * returns an Investment object representing the just purchased investment
+ */
+@WebServlet("/buy")
+public class BuyServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(BuyServlet.class.getName());
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DataSource pool = (DataSource) request.getServletContext().getAttribute("db-connection-pool");
+        Connection conn = null;
+
+        try {
+            conn = pool.getConnection();
+            conn.setAutoCommit(false); //prevent statements from being executed immediately
```

Comment:
Double check how long this mode persists, we are using a connection pool, so the connection will be reused. May be it needs to be reset at the back to true.

Reply:
Shen we reuse a connection from the pool in a new servlet it should behave the same way as a newly created connection - defaulting to setAutoCommit(true).

Reply:
at what level is the pooling at then?

Reply:
I'm not sure what you mean sorry. From what I understand a connection gets allocated from the pool when a request is made (every time a servlet gets called when we call pool.getConnection() ), and old connections are reused if they are available in the pool, but either way they get reset on the allocation so that they behave the same as a new connection. Since we get a newly allocated connection in every servlet, the settings in this one shouldn't affect the others.